### PR TITLE
Release `Microsoft.DurableTask` `2024-10-01-preview`

### PR DESCRIPTION
**Pull request by berndverst**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [x] New resource provider.
  - [ ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [x] Other, please clarify:
    - Migration of RP from Private Repo

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

This PR moves the `Microsoft.DurableTask` API version `2024-10-01-preview` to the public repository.
Please see https://github.com/Azure/azure-rest-api-specs-pr/pull/20061 for the private repositor PR and review.

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Comment by berndverst** [comment]
Reviewers please see https://github.com/Azure/azure-rest-api-specs-pr/pull/20061 which introduced `Microsoft.DurableTask` version `2024-10-01-preview` and was reviewed.

In this PR we intend to make this resource provider public with version `2024-10-01-preview` only.

**Review by torosent — APPROVED** [review]

**Review by kaibocai — APPROVED** [review]

File: specification/durabletask/DurableTask.Management/tspconfig.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+parameters:
+  "service-dir":
+    default: ""
+  "dependencies":
+    "additionalDirectories": []
+    default: ""
+emit:
+  - "@azure-tools/typespec-autorest"
+options:
```
Comment (raych1) [inline specification/durabletask/DurableTask.Management/tspconfig.yaml:9]: Please add the SDK emitters options accordingly in order to generate SDK based on TypeSpecs.

Comment (berndverst) [reply specification/durabletask/DurableTask.Management/tspconfig.yaml:10]: @raych1 I added them. Let me know if this is correct - I couldn't find the right documentation on this, so looked at other providers and what they have done.

File: specification/durabletask/resource-manager/readme.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+```yaml $(tag) == 'package-2024-10-01-preview'
+input-file:
+  - Microsoft.DurableTask/preview/2024-10-01-preview/openapi.json
+```
+
+---
+
+# Code Generation
+
+## Swagger to SDK
```
[the remark this answers is not in the document]
Comment (berndverst) [reply specification/durabletask/resource-manager/readme.md]: Done.

File: specification/durabletask/DurableTask.Management/tspconfig.yaml (hunk 1 above)
Comment (raych1) [reply specification/durabletask/DurableTask.Management/tspconfig.yaml:10]: @weidongxu-microsoft @chunyu3 @qiaozha @tadelesh @msyyc 
please review the emitter options.

Comment (msyyc) [reply specification/durabletask/DurableTask.Management/tspconfig.yaml:10]: Looks good for Python.

Comment (weidongxu-microsoft) [reply specification/durabletask/DurableTask.Management/tspconfig.yaml:10]: @raych1 

Are we allowed to enhance this doc? https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/Getting-started-with-TypeSpec-specifications.md#arm
(especially this https://github.com/Azure/typespec-azure/blob/main/eng/feeds/arm/tspconfig.yaml)

I can make a PR, if it is OK for us to update.

**Comment by berndverst** [comment]
Could someone share the documentation required to enable the SDK TypeSpec emitters required to generate the SDKs? I cannot find any reference for this.

File: specification/durabletask/DurableTask.Management/tspconfig.yaml (hunk 1 above)
Comment (raych1) [reply specification/durabletask/DurableTask.Management/tspconfig.yaml:10]: @weidongxu-microsoft 
That's document at the typespec-azure repo which will need to update I agree, feel free to update. But I will suggest user to refer to sample project for the emitter configurations. https://github.com/Azure/azure-rest-api-specs/blob/main/specification/contosowidgetmanager/Contoso.Management/tspconfig.yaml

Comment (berndverst) [reply specification/durabletask/DurableTask.Management/tspconfig.yaml:6]: @raych1 the config here is updated to match the example at https://github.com/Azure/azure-rest-api-specs/blob/main/specification/contosowidgetmanager/Contoso.Management/tspconfig.yaml

**Pull request merged**